# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1300508708 25200
#      Fri Mar 18 21:25:08 2011 -0700
# Node ID 02e48856e4860d642cc75547f31eaa19e74cdd37
# Parent  ad05e1547398850e2b384ea22c7398d566ec6b24
Add fminbnd, fminunc functions to documentation.

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,13 @@
+2010-03-18  Rik  <octave@nomad.inbox5.com>
+
+	* interpreter/nonlin.txi, interpreter/octave.texi: Add fminbnd,
+	fminunc functions to documentation.
+
 2010-03-18  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/arith.txi, interpreter/linalg.txi: Add function chop
 	to documentation.  Re-order Utility Functions and move function dot
 	from linear algebra section to be with other utility functions.
 
 2010-03-18  Rik  <octave@nomad.inbox5.com>
 
diff --git a/doc/interpreter/nonlin.txi b/doc/interpreter/nonlin.txi
--- a/doc/interpreter/nonlin.txi
+++ b/doc/interpreter/nonlin.txi
@@ -16,39 +16,47 @@
 @c along with Octave; see the file COPYING.  If not, see
 @c <http://www.gnu.org/licenses/>.
 
 @node Nonlinear Equations
 @chapter Nonlinear Equations
 @cindex nonlinear equations
 @cindex equations, nonlinear
 
+@menu
+* Solvers::
+* Minimizers::          
+@end menu
+
+@node Solvers
+@section Solvers
+
 Octave can solve sets of nonlinear equations of the form
 @tex
 $$
  f (x) = 0
 $$
 @end tex
 @ifnottex
 
 @example
 F (x) = 0
 @end example
 
 @end ifnottex
 
 @noindent
-using the function @code{fsolve}, which is based on the @sc{Minpack}
+using the function @code{fsolve}, which is based on the @sc{minpack}
 subroutine @code{hybrd}.  This is an iterative technique so a starting
-point will have to be provided.  This also has the consequence that
+point must be provided.  This also has the consequence that
 convergence is not guaranteed even if a solution exists.
 
 @DOCSTRING(fsolve)
 
-Here is a complete example.  To solve the set of equations
+The following is a complete example.  To solve the set of equations
 @tex
 $$
  \eqalign{-2x^2 + 3xy + 4\sin(y) - 6 &= 0\cr
            3x^2 - 2xy^2 + 3\cos(x) + 4 &= 0}
 $$
 @end tex
 @ifnottex
 
@@ -159,8 +167,32 @@ The Jacobian can then be used with the f
 @example
 [x, fval, info] = fsolve (@@f, [1; 2], optimset ("jacobian", "on"));
 @end example
 
 @noindent
 which gives the same solution as before.
 
 @DOCSTRING(fzero)
+
+@node Minimizers
+@section Minimizers
+@cindex local minimum
+@cindex finding minimums
+
+Often it is useful to find the minimum value of a function rather than just
+the zeroes where it crosses the x-axis.  @code{fminbnd} is designed for the
+simpler, but very common, case of a univariate function where the interval
+to search is bounded.  For unbounded minimization of a function with
+potentially many variables use @code{fminunc}.  @xref{Optimization}, for
+minimzation with the presence of constraint functions.  Note that searches
+can be made for maxima by simply inverting the objective function
+@tex
+($F_{max} = -F_{min}$).
+@end tex
+@ifnottex
+(@code{Fto_max = -Fto_min}).
+@end ifnottex
+
+@DOCSTRING(fminbnd)
+
+@DOCSTRING(fminunc)
+
diff --git a/doc/interpreter/octave.texi b/doc/interpreter/octave.texi
--- a/doc/interpreter/octave.texi
+++ b/doc/interpreter/octave.texi
@@ -595,16 +595,21 @@ Arithmetic
 Linear Algebra
 
 * Techniques Used for Linear Algebra::
 * Basic Matrix Functions::      
 * Matrix Factorizations::       
 * Functions of a Matrix::       
 * Specialized Solvers::
 
+Nonlinear Equations
+
+* Solvers::
+* Minimizers::
+
 Diagonal and Permutation Matrices
 
 * Basic Usage::          Creation and Manipulation of Diagonal and Permutation Matrices
 * Matrix Algebra::       Linear Algebra with Diagonal and Permutation Matrices
 * Function Support::     Functions That Are Aware of These Matrices
 * Example Codes::        Some Examples of Usage
 * Zeros Treatment::      The Differences in Treatment of Zero Elements
 
